core/execution/future/market.go

```diff
@@ -3382,10 +3432,14 @@ func (m *Market) submitStopOrders(
 				confirmations = append(confirmations, conf)
 			}
 		}
-
 		evts = append(evts, events.NewStopOrderEvent(ctx, v))
 	}
 
+	// Remove any referenced orders
+	for _, order := range toDelete {
+		m.CancelOrder(ctx, order.Party, order.ID, order.ID)
+	}
```

Comment:
Do we really need to do this?

Reply:
That's what the spec says:
"If set to `Order` (which should be an object containing specification for a particular order ID), triggering should override the contained order's size with however much of the referenced order has traded. Additionally, the referenced order should have any remaining unmatched portion cancelled."

Reply:
alright, didn't remembered that,